dlt/destinations/impl/sqlalchemy/sqlalchemy_job_client.py

```diff
@@ -0,0 +1,503 @@
+from typing import Iterable, Optional, Type, Dict, Any, Iterator, Sequence, List, Tuple, IO
+from types import TracebackType
+from contextlib import suppress
+import inspect
+import math
+
+import sqlalchemy as sa
+from sqlalchemy.sql import sqltypes
+
+from dlt.common import logger
+from dlt.common import pendulum
+from dlt.common.exceptions import TerminalValueError
+from dlt.common.destination.reference import (
+    JobClientBase,
+    LoadJob,
+    RunnableLoadJob,
+    StorageSchemaInfo,
+    StateInfo,
+)
+from dlt.destinations.job_client_impl import SqlJobClientBase
+from dlt.common.destination.capabilities import DestinationCapabilitiesContext
+from dlt.common.schema import Schema, TTableSchema, TColumnSchema, TSchemaTables
+from dlt.common.schema.typing import TColumnType, TTableFormat, TTableSchemaColumns
+from dlt.common.schema.utils import pipeline_state_table, normalize_table_identifiers
+from dlt.common.storages import FileStorage
+from dlt.common.json import json, PY_DATETIME_DECODERS
+from dlt.destinations.exceptions import DatabaseUndefinedRelation
+
+
+# from dlt.destinations.impl.sqlalchemy.sql_client import SqlalchemyClient
+from dlt.destinations.impl.sqlalchemy.db_api_client import SqlalchemyClient
+from dlt.destinations.impl.sqlalchemy.configuration import SqlalchemyClientConfiguration
+
+
+class SqlaTypeMapper:
```

Comment:
TypeMapper got moved to `factory` and now has ABC class (there's one more method to detect incompatible data types but I think here it is just empty).